flink-streaming-java/src/main/java/org/apache/flink/streaming/util/HDFSCopyFromLocal.java

```diff
@@ -62,4 +64,34 @@ public void run() {
 			throw asyncException.f0;
 		}
 	}
+
+	/**
+	 * Ensure that target path terminates with a new directory to be created by fs. If remoteURI does not specify a new
+	 * directory, append local directory name.
+	 * @param fs
+	 * @param localPath
+	 * @param remoteURI
+	 * @return
+	 * @throws IOException
+	 */
+	protected static URI checkInitialDirectory(final FileSystem fs,final File localPath, final URI remoteURI) throws IOException {
```

Comment:
- Can you make this private or move the code to the `run` method?
- Can you add whitespace between the arguments, `FileSystem fs, File localPath, URI remoteURI`?
- Can you make the arguments non-final?
- Can you complete the Javadocs?

Why do we need this check?